### ISSUE TITLE
Cannot set up cross project Pub/Sub subscription

### ISSUE THREAD

**Issue by yhrn** [GoogleCloudPlatform/k8s-config-connector#150]
**Describe the bug**
A Pub/Sub subscription to an "external" topic in another project fails to fulfill the "Ready" condition with the following message: `Update call failed: error applying desired state: Error creating Subscription: googleapi: Error 404: Requested project not found or user does not have access to it (project=project-that-exists). Make sure to specify the unique project identifier and not the Google Cloud Console display name.`

I even made the GSA used by Config Connector project owner in both projects and it made no difference.

**ConfigConnector Version**

1.7.0

**To Reproduce**

1. `kubectl apply -f file_with_contents_described_below.yaml.yaml`
2. `kubectl -n some-namespace describe pubsubsubscription my-subscription`

*YAML snippets:*
```yaml
apiVersion: pubsub.cnrm.cloud.google.com/v1beta1
kind: PubSubSubscription
metadata:
  name: my-subscription
  namespace: some-namespace
spec:
  ackDeadlineSeconds: 150
  messageRetentionDuration: 86400s
  retainAckedMessages: false
  topicRef:
    external: projects/project-that-exists/topics/topic-that-exists
```

**Comment by jcanseco** [comment]
Hi Mattias. Are you able to consistently reproduce this issue? I tried what you described just now, and I was able to successfully create a `PubSubSubscription` with an external reference to a topic from another project after giving the Config Connector GSA the "Project Owner" role in the topic's project. I also tried giving the Config Connector GSA the "Pub/Sub Subscriber" role instead, and that worked as well.
